### PR TITLE
Refactor Notification Telemetry to Remove Code Duplication

### PR DESCRIPTION
**Pull request by loneursid**
DCR https://dev.azure.com/microsoft/OS/_workitems/edit/41085624

Refactor the two existing NotificationTelemetryProvider set of files to remove code duplication before introducing additional providers.

### CODE REVIEW

**Review by pmpurifoy — APPROVED** [review]

**Comment by pmpurifoy** [comment]
Only minor nit is maybe change m_helper->m_telemetryHelper. At this point we have so many helpers it'd be great to clarify which ones we are using.

**Comment by loneursid** [comment]
> Only minor nit is maybe change m_helper->m_telemetryHelper. At this point we have so many helpers it'd be great to clarify which ones we are using.

Addressed, thanks

File: dev/AppNotifications/AppNotificationTelemetry.h [hunk 1]
```diff
@@ -20,16 +21,16 @@ class AppNotificationTelemetry : public wil::TraceLoggingProvider
         winrt::hresult hr,
         std::wstring const& appId) noexcept try
     {
-        if (c_maxEventLimit >= UpdateLogEventCount())
+        if (m_telemetryHelper.ShouldLogEvent())
```
Comment (hulumane) [inline dev/AppNotifications/AppNotificationTelemetry.h:24]: cool. Looks good to me. Whats the advantage of moving to a helper? Reusability for other Telemetry providers?

**Review by hulumane — APPROVED** [review]

File: dev/AppNotifications/AppNotificationTelemetry.h (hunk 1 above)
Comment (loneursid) [reply dev/AppNotifications/AppNotificationTelemetry.h:24]: We already had  this code in two separate providers and the upcoming AppNotificationBuilderTelemetryProvider will also need access to this code.

**Pull request merged**